modules/ROOT/pages/policy-ping-federate.adoc

```diff
@@ -8,88 +8,156 @@ endif::[]
 
-== Communicating to the OAuth provider through a proxy
+[source,text,linenums]
+---
+{
+    "uid":"john.doe",
+    "mail":"john.doe@example.com"
+    "token_type":"Bearer",
+    "expires_in":3600
+}
+---
+
+The fields returned by the OAuth provider are processed by the policy, propagated throughout the Mule flow, and finally exposed to the backend if the application requesting the access uses an HTTP requester.
 
-To enable the PingFederate OAuth 2.0 Token Enforcement Policy to use the Gateway proxy settings, specify the following property:
+=== Token Caching
 
-[listing]
-anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>
+After the system validates a token, that token is cached (by default) until it expires, to improve the performance of the policy. In certain cases, such as in the case of revoked tokens, you might want to minimize or even disable caching.
 
-Setting the previous property to true makes the policy use the runtime proxy settings if specified. For example:
+To control the time taken to cache the tokens before retrying a validation against the token validation endpoint, specify the following property when starting the Mule runtime engine:
 
-----
-anypoint.platform.proxy_host=localhost
-anypoint.platform.proxy_port=8080
-----
+`anypoint.platform.max_federation_expiration_time=<expiration time in seconds; specify 0 to disable caching>`
 
-== Leveraging OAuth provider information
+Alternatively, you can specify this parameter in the `wrapper.conf` file.
+ 
+To control the number of tokens that can be cached simultaneously, specify the following property when starting the Mule runtime engine:
+ 
+`anypoint.platform.ping_federate_cache_max_size=<amount of tokens to be cached; default value is 10.000>`
 
-When a token is successfully validated by the OAuth provider, the provider sends back to the policy who requested the token validation, some extra information, including fields that are PingFederate side configurable.
+Alternatively, you can specify this parameter in the `wrapper.conf` file.
 
-Token validation response example from the authorization server:
+== Configuring the Expose Headers Option
+
+If you configured the `Expose Headers` option in the policy and if the application (for example, out-of-the-box proxies) uses the HTTP requester, the properties are redirected to the backend as headers. 
+
+For each field, the original request to the protected resource is enriched with the HTTP headers using the format: X-AGW- + <key>=<value>. For the example response from the previous section, the following headers are added:
 
 [source,text,linenums]
-----
-{
-    "uid":"john.doe",
-    "mail":"john.doe@example.com",
-    "scope":["uid","mail","cn","givenName"],
-    "grant_type":"password",
-    "cn":"John Doe",
-    "realm":"/",
-    "token_type":"Bearer",
-    "expires_in":3600,
-    "givenName":"John",
-    "access_token":"fa017a0e-1bd5-214c-b19d-03efe9f9847e"
-}
-----
+---
```

Comment:
Should these be formatted as syntax (` `)?

Reply:
Don't see this in my file. I think it was removed.